### DISCUSSION TITLE
Not able to connect/correlate different microservice spans together in one trace Jaeger UI after Opentelemetry migration

### DISCUSSION THREAD

**Discussion by miPlodder** [open-telemetry/opentelemetry-java#4632] [Q&A]
Recently, I migrated from `jaeger-client` to `Opentelemetry`, but now I see jaeger not able to trace and mapping span across different microservice. I followed this https://medium.com/jaegertracing/migrating-from-jaeger-client-to-opentelemetry-sdk-bd337d796759 for the migration. 

Is there something extra that needs to be configured in order to map different call to internal microservice in single trace? (as shown in below image)

![image](https://user-images.githubusercontent.com/22195621/180465442-10b22afb-3ff3-4256-b681-0c03bb490236.png)
 

Note: It works fine in local env (windows machine) with Jaeger running, where it's not working on Jaeger UI deployed on Openshift.

**Comment by jkwatson** [comment]
have you added the jaeger propagator per that article?

**Reply by miPlodder** [reply]
Yes, It's added as per the article.

**Reply by jkwatson** [reply]
We'd definitely need more details about your services, how they're configured and how they're instrumented to answer this, then. the propagator is what makes traces propagate across service boundaries. So, it sounds like you don't have the same propagators being used in all the services.

**Reply by miPlodder** [reply]
One service is using Opentelemetry client and the other service uses Jaeger client. Is that an issue? These are spring boot Microservices.

Moreover, in the local Windows setup running "Jaeger-all-in-one.exe", I see it working properly.

Whereas in the actual staging Openshift env which has jaeger-agent, jaeger-collector, jaeger-query (not jaeger-all-in-one) as docker containers, I see this issue.

- What specific details do you want?

**Reply by jkwatson** [reply]
can you verify the right headers are being sent on requests between the services?

**Reply by miPlodder** [reply]
Can you share some details around what headers needed to be sent b/w the microservices?

**Reply by jkwatson** [reply]
That depends on whether you're using [jaeger propagation](https://www.jaegertracing.io/docs/1.36/client-libraries/#propagation-format) or [W3C propagation](https://www.w3.org/TR/trace-context/#traceparent-header). Are you seeing any tracing headers being sent on requests?

**Reply by miPlodder** [reply]
- Yep, I see tracing headers `uber-trace-id` and `traceparent` in the request headers. Below is a sample header. Can something be concluded from it, regarding my issue?
```
{
Accept=application/json, 
application/*+json,
uber-trace-id=30a2f1f6a5f66eaa08ec6bc4e293951a:3259101a27246bec:0:0, 
traceparent=00-30a2f1f6a5f66eaa08ec6bc4e293951a-3259101a27246bec-00, 
Connection=Keep-Alive, 
User-Agent=okhttp/4.9.3, 
Host=microserviceB.svc:8443, 
Accept-Encoding=gzip, 
Content-Length=0
}
```
- Moreover, which classes intercept before every request for OTEL client so that I can DEBUG the flow so that I can  see the logic why it's ignoring the headers (sample above)? Or any other way to DEBUG these sort of issues?

**Reply by jkwatson** [reply]
This headers will be picked up the propagator implementation that is configured into the receiving service. Nothing will automatically intercept the request (unless you're using the otel java agent). You'll have to make sure that the propagators are properly used in the request interceptors.
